distribution/archives/build.gradle

```diff
@@ -222,106 +219,17 @@ Closure unzipExists = { it -> new File('/bin/unzip').exists() || new File('/usr/
 //    filename, so they must be placed in different directories.
 subprojects {
   apply plugin: 'distribution'
+  apply plugin: 'elasticsearch.internal-distribution-archive-check'
+
+  distributionArchiveCheck {
+    expectedMlLicenses.add("Boost Software License - Version 1.0 - August 17th, 2003")
```

Comment:
On master this list contains Boost and 'Apache log4cxx'. Seems log4cxx is not used anymore. Should this list maintain _all_ currently used licenses for the ml project? The reason for the mismatch was that the `checkMlCppNotice` hasn't been executed for a while now I think